src/nvim/eval/funcs.c

```diff
@@ -4278,6 +4278,7 @@ static void f_has(typval_T *argvars, typval_T *rettv, FunPtr fptr)
     "nvim",
   };
 
+  const int original_shell_error = get_vim_var_nr(VV_SHELL_ERROR);
```

Comment:
- let's add a small comment
- codebase convention is `save_foo`

```suggestion
  // XXX: eval_has_provider() may shell out :(
  const int save_shell_error = get_vim_var_nr(VV_SHELL_ERROR);
```